src/page/dataverse/dataspace/dataspace.tsx

```diff
@@ -1,21 +1,41 @@
 import { useEffect, useState } from 'react'
 import { useParams } from 'react-router-dom'
 import type { Option } from 'fp-ts/Option'
-import { match, none } from 'fp-ts/Option'
+import * as O from 'fp-ts/Option'
 import { getResourceDetails } from '@/page/dataverse/dataverse'
-import type { DataverseItemDetails } from '@/page/dataverse/dataverse'
+import type { DataverseItemDetails, DataSpace } from '@/page/dataverse/dataverse'
+import { pipe } from 'fp-ts/lib/function'
+import { GovernanceDescription } from '@/page/dataverse/dataspace/governanceDescription/governanceDescription'
+import './dataspace.scss'
 
 const Dataspace = (): JSX.Element => {
   const { id } = useParams<string>()
-  const [dataspace, setDataspace] = useState<Option<DataverseItemDetails>>(none)
+  const [dataspace, setDataspace] = useState<Option<DataSpace>>(O.none)
 
   useEffect(() => {
-    id && setDataspace(getResourceDetails(id))
+    const convertToDataSpace = (resource: Option<DataverseItemDetails>): Option<DataSpace> =>
+      pipe(
+        resource,
+        O.filterMap<DataverseItemDetails, DataSpace>(value =>
+          value.type === 'dataspace' ? O.some(value as DataSpace) : O.none
+        )
+      )
+    id && setDataspace(pipe(id, getResourceDetails, convertToDataSpace))
```

Comment:
Great job for this functional code! Let's try to improve it by grouping the functions into a single sequence of functions, in a better functional style. How about this?

```suggestion
    pipe(
      O.fromNullable(id),
      O.chain(getResourceDetails),
      O.chain(v =>
        v.type === 'dataspace' ? O.some(v as DataSpace) : O.none
      ),
      setDataspace
    )
```

The last thing I don't like is the type narrowing code:

```ts
 v.type === 'dataspace' ? O.some(v as DataSpace) : O.none
```

Why not consider implementing a type guard function? Easy to implement, see:

```ts
const isDataSpace = (resource: DataverseItemDetails): resource is DataSpace =>
  resource.type === 'dataspace'
```

Okay, now this is where the magic happens. Look:

```ts
  useEffect(() => {
    pipe(
      O.fromNullable(id),
      O.chain(getResourceDetails),
      O.filter(isDataSpace),
      setDataspace
    )
  }, [id])
```

Beautiful, isn't it?

Reply:
Amazing 😉

Reply:
Great suggestions! It makes a lot more sense :)